src/solvers/dg/3d/dg.jl

```diff
@@ -2448,6 +2569,56 @@ function calc_blending_factors!(alpha, alpha_pre_smooth, u,
   end
 end
 
+"""
+    calc_loehner_indicator!(alpha, alpha_pre_smooth, u, alpha_max, alpha_min, do_smoothing, indicator_variable, thread_cache, dg::Dg3D)
+
+Computes an indicator that models underresolution within a DG element. Adapted from FEM indicator by Löhner (1987), 
+also used in the FLASH code as standard AMR indicator. Indicator uses a specified indicator_variable to 
+estimate the second derivative of the solution locally.
+http://flash.uchicago.edu/site/flashcode/user_support/flash4_ug_4p62/node59.html#SECTION05163100000000000000
+"""
+# Calculate blending factors used for shock capturing, or AMR control
+function calc_loehner_indicator!(alpha, alpha_pre_smooth, u, alpha_max, alpha_min, do_smoothing, indicator_variable, thread_cache, dg::Dg3D)
+  @assert nnodes(dg)>=3 "implementation of Loehner indicator only works for nnodes>=3 (polydeg>1)" 
+  # Calculate blending factor
+  @unpack indicator_threaded = thread_cache
+
+  Threads.@threads for element_id in 1:dg.n_elements
+    indicator = indicator_threaded[Threads.threadid()]
+
+    # Calculate indicator variables at Gauss-Lobatto nodes
+    cons2indicator!(indicator, u, element_id, nnodes(dg), indicator_variable, equations(dg))
+
+    f_wave = 0.2 #parameter to avoid small scale fluctuations influencing the indicator
+    estimate = 0.0
+    # x direction
+    for k in 1:nnodes(dg), j in 1:nnodes(dg), i in 2:nnodes(dg)-1
+      u0 = indicator[1, i, j, k]
+      up = indicator[1, i+1, j, k]
+      um = indicator[1, i-1, j, k]
+      # dirty Löhner estimate, direction by direction, assuming constant nodes
```

Comment:
Maybe this could be a (Bachelor) student thesis to figure out the effect of direction-by-direction vs. full derivaties, and using actual LGL distances in the FD derivatives? If yes, maybe you can add this to the list...

Reply:
That's a good idea!